### PR TITLE
Add a warning for %matplotlib

### LINKED ISSUE #8365: Document how to use Matplot lib widgets
Recommend usage of `%matplotlib widget` over others.
Update Wiki
Create a diagnostic message as we do for `!pip install`, users are regularly running into this.

### PR DESCRIPTION
**Pull request by rchiodo**
Fixes #8365

### CODE REVIEW

File: .vscode/launch.json [hunk 1]
```diff
@@ -229,7 +229,7 @@
                 "--ui=tdd",
                 "--recursive",
                 "--colors",
-                //"--grep", "",
```
Comment (rchiodo) [inline .vscode/launch.json:232]: dang it

**Comment by rchiodo** [comment]
Looks like this
image

**Comment by IanMatthewHuff** [comment]
Code looks fine, just a bit unsure on the overall change here. Is widget recommended over inline? Should we change our default?

**Comment by rchiodo** [comment]
> Is widget recommended over inline? Should we change our default?

Not sure what you mean by inline? If somebody is using ```%matplotlib``` (the magic) they are picking a different renderer.

**Review by DonJayamanne — APPROVED** [review]

**Comment by rchiodo** [comment]
@IanMatthewHuff were you thinking this comes up anytime the user types 'matplotlib'? 

It's only for the magic, which is a specific command for changing how matplotlib renders. If you're doing this, you're not using inline.

**Comment by IanMatthewHuff** [comment]
> @IanMatthewHuff were you thinking this comes up anytime the user types 'matplotlib'?
> 
> It's only for the magic, which is a specific command for changing how matplotlib renders. If you're doing this, you're not using inline.

It's picking the renderer, and [inline] is a renderer option. We don't want a squiggly for that do we?

**Comment by rchiodo** [comment]
> It's picking the renderer, and [inline] is a renderer option. We don't want a squiggly for that do we?

I didn't even notice inline was an option. A redundant option but nevertheless we should skip the warning on that one.

**Comment by IanMatthewHuff** [comment]
> > It's picking the renderer, and [inline] is a renderer option. We don't want a squiggly for that do we?
> 
> I didn't even notice inline was an option. A redundant option but nevertheless we should skip the warning on that one.

Sure that make sense. The initial change looked like a partial attempt to replace one with the other. If you exclude inline as well then I'm fine with this.

**Review by IanMatthewHuff — APPROVED** [review]
Approved if 'inline' is excluded from squiggle warning.

**Comment by rchiodo** [comment]
> Approved if 'inline' is excluded from squiggle warning.

Actually I just tried 'inline'. It crashes the kernel. Seems it's not the default. Not sure what it is.

**Comment by rchiodo** [comment]
Okay never mind, that kernel is just hosed. Another kernel, inline is the default and should work.

**Pull request merged**